main_f4.c

```diff
@@ -54,7 +54,57 @@ static struct {
 #define OTP_SIZE			512
 #define UDID_START		        0x1FFF7A10
 
+// address of MCU IDCODE
+#define DBGMCU_IDCODE		0xE0042000
+#define STM32_UNKNOWN	0
+#define STM32F40x_41x	0x413
+#define STM32F42x_43x	0x419
+#define STM32F42x_446xx	0x421
+
+#define REVID_MASK	0xFFFF0000
+#define DEVID_MASK	0xFFF
+
+/* magic numbers from reference manual */
+
+typedef enum mcu_rev_e {
+	MCU_REV_STM32F4_REV_A = 0x1000,
+	MCU_REV_STM32F4_REV_Z = 0x1001,
+	MCU_REV_STM32F4_REV_Y = 0x1003,
+	MCU_REV_STM32F4_REV_1 = 0x1007,
+	MCU_REV_STM32F4_REV_3 = 0x2001
+} mcu_rev_e;
+
+typedef struct mcu_des_t {
+	uint16_t mcuid;
+	const char *desc;
+	char  rev;
+} mcu_des_t;
+
+// The default CPU ID  of STM32_UNKNOWN is 0 and is in offset 0
+// Before a rev is known it is set to ?
+// There for new silicon will result in STM32F4..,?
+mcu_des_t mcu_descriptions[] = {
+	{ STM32_UNKNOWN,	"STM32F???",    '?'},
+	{ STM32F40x_41x, 	"STM32F40x",	'?'},
+	{ STM32F42x_43x, 	"STM32F42x",	'?'},
+	{ STM32F42x_446xx, 	"STM32F446XX",	'?'},
+};
+
+typedef struct mcu_rev_t {
+	mcu_rev_e revid;
+	char  rev;
+} mcu_rev_t;
+
+const mcu_rev_t silicon_revs[] = {
+	{MCU_REV_STM32F4_REV_3, '3'}, /* Revision 3 */
 
+	{MCU_REV_STM32F4_REV_A, 'A'}, /* Revision A */  // FIRST_BAD_SILICON_OFFSET (place good ones above this line)
```

Comment:
@davids5 : Perhaps add a bit more to this comment. Add to also update the FIRST_BAD_SILICON_OFFSET.
e.g.
```
// FIRST_BAD_SILICON_OFFSET (place good ones above this line and update the FIRST_BAD_SILICON_OFFSET accordingly)
``